src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,568 @@
+                 {isFilterOpen ? <img src={filterOpen} alt={filterOpen}/> : <img src={filter} alt={filter}/>}
+             </div>)
+     }
+ 
+     /*Function to display multiselect Header*/
+     const renderMultiSelectDataGridHeader = () => {
+         return (<th scope="col">
+                 <div>
+                     <label>
+                         <input type="checkbox" id="allSelect"
+                                checked={allValues.filter((value: any) => value?.isChecked !== true).length < 1}
+                                onChange={handleMultiSelect}/>
+                     </label>
+                 </div>
+             </th>)
+     }
+ 
+     /*Function to display multiselect rows*/
+     const renderMultiSelectDataGridRow = (row: any) => {
+         return (<th scope={row}>
+                 <div>
+                     <label>
+                         <input type="checkbox" id={row.id}
+                                onChange={handleMultiSelect}
+                                checked={row?.isChecked || false}/>
+                     </label>
+                 </div>
+             </th>)
+     }
+ 
+     /*Function to render Single Select Row*/
+     const renderSingleSelectDataGridRow = (row: any) => {
+         return (<th>
+                 <div>
+                     <input type="radio"
+                            name="radio-group"
+                            id="radio-button-control" value={row}
+                            onClick={() => handleRadioSelection(row)}
+                     />
+                 </div>
+             </th>)
+     }
+ 
+     /**
+      * renderEmptyDatagrid function returns the empty datagrid message and images in the table body
+      *
+      */
+     const renderEmptyDatagrid = () => {
+         return (<div className="empty-datagrid-style">
+                 <div className="datagrid-placeholder-container">
+                     <div
+                         className="datagrid-placeholder datagrid-empty clr-align-items-center clr-justify-content-center">
+                         <div/>
+                         <img src={emptyDatagrid} alt={emptyDatagrid}/>
+                         {Constants.EMPTY_DATA_GRID_NO_ITEMS_FOUND}
+                     </div>
+                 </div>
+             </div>)
+     }
+ 
+     /**
+      * renderColumnSelection function returns the column selection
+      */
+     const renderColumnSelection = () => {
+         return (<div className="column-switch clr-popover-content"
+                      ref={refChild}
+                      role="dialog"
+                      style={{
+                          top: 0, bottom: "auto", right: "auto", left: 0, transform: transformVal,
+                      }}>
+                 <div className="ColumnSelect">
+                     <span className="ColumnSelect-header">{Constants.COLUMN_PICKER}</span>
+ 
+                     {allColumns.map((column: any) => <div key={column.id}>
```

Comment:
any -> specific type

Reply:
Have added the TODO comment, will take care to this while writing wrappers.